dora/core/server/worker/src/main/java/alluxio/worker/netty/BlockReadRequest.java

```diff
@@ -30,7 +32,7 @@ public final class BlockReadRequest extends ReadRequest {
    */
   BlockReadRequest(Protocol.ReadRequest request) {
     super(request.getBlockId(), request.getOffset(), request.getOffset() + request.getLength(),
-        request.getChunkSize());
+        downcastChunkSize(request));
```

Comment:
I don't understand why this is needed.

```
class ReadRequest {
  private final long mId;
  private final long mStart;
  private final long mEnd;
  private final long mPacketSize;
  private final long mSessionId;

  ReadRequest(long id, long start, long end, long packetSize) {
    mId = id;
    mStart = start;
    mEnd = end;
    mPacketSize = packetSize;
    mSessionId = IdUtils.createSessionId();
  }

```

The constructor expects a long for packetSize.
Why do we need to downcast it to int?

Reply:
it is confusing to me that:
Now we are doing "page based reading" in Dora.
But this is called "block read request".
I know the Netty implementation is picked from old version.
But do you think it is a good time to change the terminologies?

Reply:
I changed that to `int`. https://github.com/Alluxio/alluxio/pull/17479/files#diff-0e33aad17b8460eb5b842aff737b0c63c189038229f77a5c1a3d657f9c3c4880

A packet is the transfer unit of a read request over the Netty channel. The read request may request a range larger than `int` can represent, but we are never going to have a packet size that exceeds `Integer.MAX_VALUE` which is roughly 2 GiB. On the other hand, we need the packet size to allocate buffers, compute offsets in buffer, etc., and those APIs require an `int` as the length. See https://github.com/Alluxio/alluxio/blob/aa8e38cfa7ebac392f68fad949cc4e344e2573f8/dora/core/server/worker/src/main/java/alluxio/worker/netty/NettyReadHandlerStateMachine.java#L646-L653

Reply:
> do you think it is a good time to change the terminologies

Yes, Jiacheng is removing all block-based APIs from the Dora codebase in #17638 . This PR predates his, so I had to keep the compatibility.

Reply:
OK